### ISSUE TITLE
Remove all uses of deferredGenerator.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6626]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6626|
|Type|enhancement|
|Created|2013-07-14 22:37:39Z|
|Branch|https://github.com/twisted/twisted/tree/remove-use-of-deferredGenerator-6626-2|

This is a prerequisite of deprecating it in [#6044](https://github.com/twisted/twisted/issues/6044).

All but one of the uses are in tests.

<details><summary>Searchable metadata</summary>

```
trac-id__6626 6626
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_remove_use_of_deferredGenerator_6626_2 branches/remove-use-of-deferredGenerator-6626-2
branch_author__tom_prince__tomprince tom.prince, tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1373841459000000 1373841459000000
changetime__1397526818000000 1397526818000000
version__None None
owner__tom_prince tom.prince
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @tomprince|
|-|-|

Hi Tom,

found some more references.

```
./twisted/trial/util.py:314:_runSequentially = defer.deferredGenerator(_runSequentially)
./twisted/trial/test/test_runner.py:893:        test_spam = defer.deferredGenerator(test_bar)
./twisted/conch/test/test_manhole.py:281:    testControlD = defer.deferredGenerator(testControlD)
./twisted/conch/test/test_manhole.py:304:    testControlL = defer.deferredGenerator(testControlL)
./twisted/conch/test/test_manhole.py:359:    testDeferred = defer.deferredGenerator(testDeferred)
./twisted/test/test_newcred.py:16:from twisted.internet.defer import deferredGenerator as dG, waitForDeferred as wFD
./twisted/test/test_ftp.py:387:    testPASV = defer.deferredGenerator(testPASV)
```

They should be replaced as well no? (you manually set the branch so i hope im looking at the right one: `remove-use-of-deferredGenerator-6626`)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince commented|
|-|-|

That looks like the list of changes I made in the branch. Looking at github, it looks like the correct one. (diffresource is confused, though)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

I clicked the (trac) log link at the top of this ticket and clicked view changes ([this link](https://twistedmatrix.com/trac/changeset?new=39170%40branches%2Fremove-use-of-deferredGenerator-6626&old=39168%40branches%2Fdeprecate-deferredGenerator-6044-2)) and it only shows changes in 2 files. So that link isn't reliable?

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

> (diffresource is confused, though)

The problem is the branch was made outside of the workflow and it's not obvious how to get the changes it contains.  If you can, please re-create the branch following the standard workflow so that the normal expectations of developers and tools are satisfied.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_9"></a>|@tomprince commented|
|-|-|

(In [39255]) Branching to remove-use-of-deferredGenerator-6626-2.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @tomprince|
|-|-|

Thanks.  These changes look good to me.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince set status to `closed`|
|-|-|

(In [42221]) Merge remove-use-of-deferredGenerator-6626-2: Remove all uses of deferredGenerator.

Author: tom.prince
Reviewers: thijs, exarkun
Fixes: [#6626](https://github.com/twisted/twisted/issues/6626)
